File: src/main/java/cn/gov/forestry/executor/client/AssetsInnerResourceClient.java
package cn.gov.forestry.executor.client;

import cn.gov.forestry.common.file.FileContent;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(name = "assetsInnerResourceClient", url = "${forim.inner.assets.url}")
public interface AssetsInnerResourceClient {
    /**
     * name字段带相对路径,如/sprite/a.png
     * 实际存储路径assets模块会拼接上systemId区分开,所以请求时只需要按照各个业务模块区分就行
     * */
    @PostMapping(value = "/inner/resource/put/file")
    FileContent putResourceFile(@RequestBody FileContent fileContent);

    @PostMapping(value = "/inner/resource/get/file")
    FileContent getResourceFile(@RequestBody FileContent fileContent);
}

File: src/main/python/cn/gov/forestry/executor/client/AssetsInnerResourceClient.py
"""Python adaptation of the Java `AssetsInnerResourceClient` interface.

Methods mirror the Java naming style (camelCase):
- putResourceFile(FileContent) -> FileContent
- getResourceFile(FileContent) -> FileContent
"""
from typing import Optional
import os
import requests

from cn.gov.forestry.common.file.file_content import FileContent


class AssetsInnerResourceClient:
    def __init__(self, base_url: Optional[str] = None, timeout: int = 30):
        self.base_url = base_url or os.getenv('FORIM_INNER_ASSETS_URL')
        if not self.base_url:
            raise ValueError('base_url must be provided or FORIM_INNER_ASSETS_URL set')
        self.timeout = timeout

    def _post(self, path: str, payload: dict) -> dict:
        url = self.base_url.rstrip('/') + path
        resp = requests.post(url, json=payload, timeout=self.timeout)
        resp.raise_for_status()
        return resp.json()

    def putResourceFile(self, fileContent: FileContent) -> FileContent:  # Java-style name
        data = fileContent.to_dict()
        body = self._post('/inner/resource/put/file', data)
        return FileContent.from_dict(body)

    def getResourceFile(self, fileContent: FileContent) -> FileContent:  # Java-style name
        data = fileContent.to_dict()
        body = self._post('/inner/resource/get/file', data)
        return FileContent.from_dict(body)


__all__ = ['AssetsInnerResourceClient']
